src/go/rpk/pkg/out/out.go

```diff
@@ -200,3 +202,23 @@ func (t *TabWriter) PrintColumn(header string, args ...interface{}) {
 func (t *TabWriter) Line(sprint ...interface{}) {
 	fmt.Fprint(t.Writer, append(sprint, "\n")...)
 }
+
+// StructuredPrint marshalls and prints the structToPrint based on the format type.
+func StructredPrint[T any](structToPrint T, format string) {
```

Comment:
Typo in Structured

The generic here doesn't really add anything -- there's no method being used in the generic, and this isn't a container. I'd just make this:
```go
func StructuredPrint(v any, format string)
```
Also, since this is doing a formatted print (as implied by the arg), I'd use `FormattedPrint`, and to be consistent with the functions above, I'd actually use `PrintFormatted`.

Reply:
I thought generics were required to support this kind use case, I'm still learning as I go. I'll remove it and rename the function as suggested.